### ISSUE TITLE
Closing backtick followed by apostrophe does not end code block

### ISSUE THREAD

**Issue by lkysow** [asciidoctor/asciidoctor-intellij-plugin#699]
## Observed vs. expected behavior

This doc:
```
`test`'s `formatting` doesn't work when it should
```

Results in:
![image](https://user-images.githubusercontent.com/1034429/110899625-4f3d2c80-82b6-11eb-86e6-75152a88a47f.png)

## Steps to reproduce

Use a string that ends a code block with a backtick immediately followed by an apostrophe.

## Environment

Plugin Version: 0.32.26

IntelliJ Details:
GoLand 2020.3.2
Build #GO-203.7148.71, built on January 26, 2021

**Comment by ahus1** [comment]
Thanks for reporting this. Reading your bug report, I assume that `test` and `formatting` should be code blocks, and the result is taken from the preview.

TL;DR: The preview is correct and the syntax highlighting in the editor is wrong. Please use double-backticks around `test` like this:

```
``test``'s `formatting` doesn't work when it should
```

The longer explanation: 

What you see is AsciiDoc's constrained vs. unconstrained formatting: when using single backticks, the need to be surrounded by spaces or punctation marks. When marking something inside a word, use double backticks. See [constrained vs. unconstrained formatting](https://docs.asciidoctor.org/asciidoc/latest/text/) in the docs.

There is another special case in this example: <code>`'</code> is special two-character combination for a [curved quote](https://docs.asciidoctor.org/asciidoc/latest/subs/quotes/).

So what you see in the preview is the correct AsciiDoc behavior. 

Unfortunately there is still a bug here, as the example text isn't correctly shown in the editor with syntax highlighting.

![image](https://user-images.githubusercontent.com/3957921/110909364-4da05280-8310-11eb-8aa9-5a918f980829.png)

I need to update the plugins lexer to support this. I see that also the quick-format actions in the editor's toolbar have an issue when the formatting leads to a <code>`'</code>.

Reason for two separated lexers in the editor and the preview is that the parsing AsciiDoc is using for rendering is not compatible with the parsing needed for an IDE's editor. The aim is for the AsciiDoc Language Project at Eclipse to come up with a grammar that works for both a renderer and an editor.

**Comment by lkysow** [comment]
Hi, thanks so much for your detailed and quick response. That makes sense! My file is actually getting processed by another system which renders it as I had previously expected so that's why I thought the plugin was wrong but it looks like that other system actually has it incorrect.

Feel free to update my ticket to better describe the actual bug.

**Comment by ahus1** [comment]
There's now pre-release 0.32.29 that provides better lexing and parsing, and fixed actions in the editor's toolbar. Please give it a try. 

As you mentioned that you use another system that renders the output, maybe that system is not using Asciidoctor to render the output. I'm not sure if all AsciiDoc implementations support the typographic quote <code>`'</code>.

You'll see that the editor will not yet show the typographic quote within monospaced text with a grey background; that's something for a future release.

![image](https://user-images.githubusercontent.com/3957921/111071765-f6e57500-84d7-11eb-9b74-284d285d2d43.png)
 

The pre-release of the plugin is available from [GitHub releases](https://intellij-asciidoc-plugin.ahus1.de/docs/contributors-guide/user/installing-a-preview-release.html) and the [IntelliJ AsciiDoc EAP repository](https://intellij-asciidoc-plugin.ahus1.de/docs/contributors-guide/user/add-eap-repository-to-ide.html).

**Comment by ahus1** [comment]
Syntax highlighting in the editor for elements nested inside of monospace or italic will work starting with the next pre-release. I'll update this issue once it is available.

**Comment by ahus1** [comment]
Pre-Release 0.32.30 is available that includes these changes.

The pre-release of the plugin is available from [GitHub releases](https://intellij-asciidoc-plugin.ahus1.de/docs/contributors-guide/user/installing-a-preview-release.html) and the [IntelliJ AsciiDoc EAP repository](https://intellij-asciidoc-plugin.ahus1.de/docs/contributors-guide/user/add-eap-repository-to-ide.html).
